tests/test.linter.js

```diff
@@ -183,6 +183,38 @@ describe('Linter', function() {
       });
   });
 
+  it('Eslint ignore patterns and .eslintignorerc should be ignored', () => {
+    // Verify https://github.com/mozilla/addons-linter/issues/1288 is fixed
+    var addonLinter = new Linter({_: [
+      'tests/fixtures/webextension_node_modules_bower']});
+
+    // Stub print to prevent output.
+    addonLinter.print = sinon.stub();
+
+    assert.equal(addonLinter.collector.warnings.length, 0);
+    assert.equal(addonLinter.collector.warnings.length, 0);
+
+    var getFileSpy = sinon.spy(addonLinter, 'scanFile');
+
+    return addonLinter.scan()
+      .then(() => {
+        // Ensure no warnings are raised. If we would not overwrite eslint
+        // ignore patterns eslint ignores `node_modules` and
+        // `bower_components` by default. We want them to be scanned though.
+        assert.equal(addonLinter.collector.warnings.length, 0);
```

Comment:
It just isn't clear to me why `warnings.length` still being `0` proves those folders were scanned. Nothing has changed. Wouldn't you want a file with a warning inside the `node_modules` folder so we can ensure it's scanned and raises an error?